services/src/main/java/org/exoplatform/news/NewsServiceImpl.java

```diff
@@ -1210,6 +1210,20 @@ public String formatMention(String text) {
           next = "<a href=\"" + CommonsUtils.getCurrentDomain() + identity.getProfile().getUrl() + "\">"
                   + encoder.encodeHTML(identity.getProfile().getFullName()) + "</a>";
         }
+      } else if (next.startsWith("<p>@")) {
+        String username = next.substring(4);
+        org.exoplatform.social.core.identity.model.Identity identity = loadUser(username);
+        if (identity != null) {
+          next = "<p><a href=\"" + CommonsUtils.getCurrentDomain() + identity.getProfile().getUrl() + "\">"
+                  + encoder.encodeHTML(identity.getProfile().getFullName()) + "</a>";
+        }
+      } else if (next.contains("@")) {
+        String username = next.split("@")[1];
+        org.exoplatform.social.core.identity.model.Identity identity = loadUser(username);
+        if (identity != null) {
+          next = next.split("@")[0] + "<a href=\"" + CommonsUtils.getCurrentDomain() + identity.getProfile().getUrl() + "\">"
+                  + encoder.encodeHTML(identity.getProfile().getFullName()) + "</a>";
+        }
```

Comment:
Could we combine the 3 checks in one and just remove all characters before the @ ? we can imagine having other HTML tags before the @ character